src/server/main_service.cc

```diff
@@ -584,6 +584,29 @@ void Service::Shutdown() {
   ThisFiber::SleepFor(10ms);
 }
 
+bool Service::CheckKeysOwnership(const CommandId* cid, CmdArgList args,
+                                 ConnectionContext* dfly_cntx) {
+  if (cid->first_key_pos() == 0) {
+    return true;  // No key command.
+  }
+  OpResult<KeyIndex> key_index_res = DetermineKeys(cid, args);
+  if (!key_index_res) {
+    (*dfly_cntx)->SendError(key_index_res.status());
+    return false;
+  }
+
+  const auto& key_index = *key_index_res;
+  for (unsigned i = key_index.start; i < key_index.end; ++i) {
+    string_view key = ArgS(args, i);
+    if (!server_family_.cluster_data()->IsKeyInMySlot(key)) {
+      VLOG(1) << "Key " << key << " is not owned by server";
+      (*dfly_cntx)->SendError("MOVED");  // TODO add more info to moved error.
```

Comment:
Let's think of the case in which a command was issued on multiple keys, and *some* are owned by this instance, but others are not.
In this case, I don't think that the proper response is `MOVED`, because there's no way to issue that command no matter which instance receives it.

Reply:
Actually, any command which runs on multiple instances can't be issued, and we also can't return `MOVED` for it (we won't be able to redirect the response to any specific instance).
As such, perhaps we should change the `IsKeyInMySlot()` to be `GetSlotForKey(key)`, which returns a slot id.
Then we can determine the response:
1. If 1 instance, and it is the current instance: handle it as usual
2. If 1 instance but it's _not_ the current one: return `MOVED`
3. If it's >1: return an error

Reply:
actually, I will block requests on different slots, as redis does. 
So I will check first if keys belong to different slots and than if this slot belongs to me